Add more RST docs for optimisation module (#2413)
Fusion-Power-Plant-Framework/bluemira: documentation/source/optimisation/geometry.rst

        """Geometry parameterisation for a circle in the xz-plane."""

        def __init__(
            self, var_dict: Optional[Dict[str, Union[float, Dict[str, float]]]] = None
        ):
            opt_variables = OptVariables(
                [
                    BoundedVariable("radius", 10, 1e-5, 15),
                    BoundedVariable("centre_x", 0, -10, 10),
                    BoundedVariable("centre_z", 0, 0, 10),
                ]
            )
            opt_variables.adjust_variables(var_dict, strict_bounds=True)
            super().__init__(opt_variables)

        def create_shape(self, label: str = "") -> BluemiraWire:
            """Create the circle."""
            return make_circle(
                self.variables["radius"].value,
                center=(
                    self.variables["centre_x"].value,
                    0,
                    self.variables["centre_z"].value,
                ),
                axis=(0, 1, 0),
                label=label,
            )

.. note::
    For more information on
    :py:class:`~bluemira.geometry.parameterisations.GeometryParameterisation`\s
    see :doc:`here <../geometry/parameterisation>`.

> Next step is to create our zone in which our circle must remain outside.

Reviewer: ```suggestion
Next step is to create a zone our circle must remain outside.
```
